### ISSUE TITLE
relax unit_ids to not have to be integer?

### ISSUE THREAD

**Issue by eackermann** [eackermann/nelpy#94]
Alternatively, we should re-think how to showcase our raster flexibility against that of matplotlib's eventplot()

**Comment by eackermann** [comment]
But it is nice to have some expectation about what (mostly hidden) unit_ids are, and the unit_labels are really there for users to call whatever they want; so I don't want to support arbitrary types. Does float really make that much more sense than int? Would some people _really_ want something more flexible? One possibility is to have unit 1a and 1b if you think they're related, or even if you think it's the same unit, but before applying some chemical intervention, and after, for example.

**Comment by jchutrue** [comment]
I think it makes more sense for the id to be a unique integer identifier (consistent with other uses I've seen of what an id means). Users can always custom label the cells 1a, 1b, etc. if they're so inclined.

**Comment by eackermann** [comment]
I agree. Uniqueness is already enforced using set() I believe. So then we're saying most probably we just want to support those *possible* custom offsets?

**Comment by eackermann** [comment]
Also, if your parents are in town, you should spend time with them!!!!! #

**Comment by jchutrue** [comment]
I'm wondering though if having custom offsets will make it more difficult to understand the plot. A constant  vertical length between units is easier to look at in my opinion. But perhaps I haven't seen the cases where a custom offset would be useful.

**Comment by eackermann** [comment]
In principal, I agree with you. But http://matplotlib.org/examples/pylab_examples/eventplot_demo.html though it necessary to include and showcase this functionality. The whole point of this issue is to show that we can fully re-create anything in the existing raster, and more!
